File: 2411-smallest-subarrays-with-maximum-bitwise-or/2411-smallest-subarrays-with-maximum-bitwise-or.java
class Solution {
    public int[] smallestSubarrays(int[] nums) {
        int n = nums.length;
        int[] last = new int[32];
        
        int[] answer = new int[n];
        Arrays.fill(answer,1);
        
        for(int i=n-1;i>=0;i--){
            
            for(int j=0;j<32;j++){
                if((nums[i] & (1<<j))>0){
                    last[j] = i;
                }
                
                answer[i] = Math.max(answer[i],last[j]-i+1);
                
            }
        }
        
        return answer;
        
        
            
            
        
    }
}

File: 2411-smallest-subarrays-with-maximum-bitwise-or/2411-smallest-subarrays-with-maximum-bitwise-or.ts
function smallestSubarrays(nums: number[]): number[] {
    
    let n = nums.length;
    
    let answer: number[] = Array(n).fill(1)
    
    let last : number[] = Array(32).fill(0)
    
    for(let i=n-1;i>=0;i--){
        for(let j=0;j<32;j++){
            if(nums[i] & (1<<j)){
                last[j] = i
            }
            
            answer[i] = Math.max(answer[i],last[j]-i+1)
        }
    }
    
    return answer
    
};